### PR TITLE
Add predefined_recognizer: IN_AADHAAR

### PR DESCRIPTION
**Pull request by devopam**
## Change Description
Add a new recognizer - IN_AADHAAR, language: en

Describe your changes
Add Govt of India issued unique person identifier : Aadhaar

## Issue reference

This PR fixes issue #XX

## Checklist

- [X] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [X] I have signed the CLA (if required)
- [X] My code includes unit tests
- [X] All unit tests and lint checks pass locally
- [X] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by devopam** [comment]
I would like to know where to set the default score for a newly introduced recognizer.

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+    for different strings to be used during pattern matching.
+    This can allow a greater variety in input, for example by removing dashes or spaces.
+    """
+
+    PATTERNS = [
+        Pattern(
+            "AADHAAR (Low)",
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py:23]: As this is a very weak detector, I would suggest changing the pattern name to AADHAAR (very week) and the default score to 0.01

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            supported_language=supported_language,
+        )
+
+    def validate_result(self, pattern_text: str) -> bool:
+        """Determine absolute value based on calculation."""
+        sanitized_value = self.__sanitize_value(pattern_text, self.replacement_pairs)
+        return self.__check_aadhaar(sanitized_value)
+
+    @staticmethod
+    def __check_aadhaar(sanitized_value: str) -> bool:
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py:62]: Could you please provide a reference to this validation logic? A quick search mentions the Verhoeff Algorithm. Is this the same?

**Review by omri374 — COMMENTED** [review]
Thanks for the contribution! Left a couple of comments

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py (hunk 1 above)
Comment (devopam) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py]: thanks for the suggestion. implemented as desired.

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py (hunk 2 above)
Comment (devopam) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py:62]: this is the most detailed article that can be used for referencing. While the knowledge is available in public domain , it is not acknowledged officially in the goverment documents. https://medium.com/@krs.sharath03/how-aadhar-number-is-generated-and-validated-3c3e7172e606

**Comment by devopam** [comment]
hi @omri374 ,
I added a new python library : stdnum for use. Will it automatically get added to requirements.txt or I need to manually refer it somewhere ? Please advise

**Comment by omri374** [comment]
Hi @devopam unfortunately we cannot use stdnum because of its license. Are you familiar with another implementation with a less strict license?

**Comment by devopam** [comment]
Hi @omri374 ,
Nice catch on stdnum , I checked their licensing and it's actually a bit conflicting (scripts vs LGPL on pip) but nonetheless it is out of my purview. verhoeff is a standard algo, i will add it as a function in main class itself and re-submit.

**Comment by omri374** [comment]
I think that's a good way forward. Perhaps create a `checksum` or `validation` file with the verhoeff algorithm? We may add more to it in the future.

**Comment by devopam** [comment]
Hi @omri374 ,

I have added analyzer_utils.py as a generic class for utility functions, and created the logic for verhoeff validation in it as a function.  Please have a look and let me know .

**Review by omri374 — COMMENTED** [review]
Thanks! Looks great. Left one comment asking to add a few unit tests to the utils class.

File: presidio-analyzer/presidio_analyzer/analyzer_utils.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        :param text: input string
+        :param replacement_pairs: pairs of what has to be replaced with which value
+        :return: cleansed string
+        """
+        for search_string, replacement_string in replacement_pairs:
+            text = text.replace(search_string, replacement_string)
+        return text
+
+    @staticmethod
+    def is_verhoeff_number(input_number: int):
```
[the remark this answers is not in the document]
Comment (devopam) [reply presidio-analyzer/presidio_analyzer/analyzer_utils.py:40]: @omri374  oops, forgot that . 
Thank you for reviewing the same and pointing it out. 
Added the same. Please have a look. 
regards Devopam

**Review by omri374 — DISMISSED** [review]

**Comment by omri374** [comment]
no worries :)

**Review by omri374 — APPROVED** [review]
Thanks!

**Pull request merged**